alvarocasadoc/day1/day1.exs

```diff
@@ -0,0 +1,22 @@
+# AdventOfCode2018
+# @alvarocasadoc
+# Day 1
+
+defmodule Day1 do
+
+  # Part 1
+  def sum_frequencies(input) do
+    items = Enum.map(input, fn x -> Integer.parse(x) |> elem(0) end)
+    Enum.sum(items)
+  end
+
+  # Part 2
+  def find_first_cycle(input) do
+    items = Enum.map(input, fn x -> Integer.parse(x) |> elem(0) end)
+    stream = Stream.scan(Stream.cycle(items), &(&1 + &2))
+
+    Enum.reduce_while(stream, [], fn current, prev ->
+      if current in prev, do: {:halt, current}, else: {:cont, [current | prev]}
+    end)
```

Comment:
Hmmm this kind of syntax is why I don't like Ruby or Elixir that much (though they are pretty cool).

It is pretty clear what this is doing, but still, what are `:halt` and `:cont` exactly? I suppose this is some kind of trick they are using internally but I would prefer some other way of expressing I want to stop or continue. Maybe a lambda or a keywork like `return` dunno.

Anyway, this function is pretty clever. I would say this is even more declarative than stopping recursion in Haskell 😇